neurodamus/utils/version.py

```diff
@@ -0,0 +1,62 @@
+"""git_version.py
+
+Provides utilities for handling Git-style version strings (e.g., '9.0a-1485-g0d990513b').
+
+Includes:
+- GitVersion: class representing a version with base, commit count, and hash,
+  supporting comparison operators.
+- check_program_version: function to verify a program meets a minimum Git-version requirement.
+
+Intended for use with programs that use Git-describe-style versioning, such as NEURON.
+"""
+
+import re
+from functools import total_ordering
+
+from packaging import version
```

Comment:
this ads a new dependency, why not just use the regex to pull the first part?

Reply:
either I lose correct handling of alpha/beta/release candidate suffixes (9.0a, 9.0b1) or I reinvent the wheel to do exactly what they doin packaging. If you prefer, yes, we can do that

Reply:
> Should I add to requirements.txt?

Additional dependencies should go in pyproject.toml.
I'm not sure why there is a requirements.txt; it's not doing anything.